tests/generator/bug57531.cs

```diff
@@ -0,0 +1,23 @@
+using System;
+using UIKit;
+using Foundation;
+using ObjCRuntime;
+using CoreGraphics;
+using CoreLocation;
+using Contacts;
+
+namespace Bug57531 {
+
+	[Category]
+	[BaseType (typeof (CLGeocoder))]
+	interface CLGeocoder_ContactsAdditions {
+
+		[Export ("geocodePostalAddress:completionHandler:")]
+		[Async]
+		void GeocodePostalAddress (CNPostalAddress postalAddress, CLGeocodeCompletionHandler completionHandler);
+
+		[Export ("geocodePostalAddress:preferredLocale:completionHandler:")]
+		[Async]
+		void GeocodePostalAddress (CNPostalAddress postalAddress, [NullAllowed] NSLocale locale, CLGeocodeCompletionHandler completionHandler);
+	}
```

Comment:
Those were inlined, do you recall of any other cases that were not ?

Test ensure it compiles (which is fine for a generator test), however it does not mean the generated code work (I assume it does, but if there were any other case to fix in the bindings then adding a unit test would confirm it)/

Reply:
The generated code is calling the not async version of it so as long the non async works the API should work as the other existing APIs here is a snippet of the generated code:

```csharp
[Export ("geocodePostalAddress:preferredLocale:completionHandler:")]
[CompilerGenerated]
public unsafe static void GeocodePostalAddress (this global::CoreLocation.CLGeocoder This, global::Contacts.CNPostalAddress postalAddress, NSLocale locale, [BlockProxy (typeof (ObjCRuntime.Trampolines.NIDCLGeocodeCompletionHandler))]global::CoreLocation.CLGeocodeCompletionHandler completionHandler)
{
	if (postalAddress == null)
		throw new ArgumentNullException ("postalAddress");
	if (completionHandler == null)
		throw new ArgumentNullException ("completionHandler");
	BlockLiteral *block_ptr_completionHandler;
	BlockLiteral block_completionHandler;
	block_completionHandler = new BlockLiteral ();
	block_ptr_completionHandler = &block_completionHandler;
	block_completionHandler.SetupBlockUnsafe (Trampolines.SDCLGeocodeCompletionHandler.Handler, completionHandler);
	
	global::ApiDefinition.Messaging.void_objc_msgSend_IntPtr_IntPtr_IntPtr (This.Handle, Selector.GetHandle ("geocodePostalAddress:preferredLocale:completionHandler:"), postalAddress.Handle, locale == null ? IntPtr.Zero : locale.Handle, (IntPtr) block_ptr_completionHandler);
	block_ptr_completionHandler->CleanupBlock ();
	
}

[CompilerGenerated]
public unsafe static Task<global::CoreLocation.CLPlacemark[]> GeocodePostalAddressAsync (this global::CoreLocation.CLGeocoder This, global::Contacts.CNPostalAddress postalAddress, NSLocale locale)
{
	var tcs = new TaskCompletionSource<global::CoreLocation.CLPlacemark[]> ();
	This.GeocodePostalAddress(postalAddress, locale, (placemarks_, error_) => {
		if (error_ != null)
			tcs.SetException (new NSErrorException(error_));
		else
			tcs.SetResult (placemarks_);
	});
	return tcs.Task;
}
```

Now answering your actual question

> do you recall of any other cases that were not ?

Not really, if you want I can create a small sample testing this API, since it is an extension method it can be tested in a third party binding.

Reply:
no, just wondered if another commit was coming with fixes to check the generated code/tests :)